dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-completed-fee-record-corrections.controller/helpers/map-completed-corrections-to-response.ts

```diff
@@ -0,0 +1,38 @@
+import { FeeRecordCorrectionEntity, mapReasonsToDisplayValues } from '@ukef/dtfs2-common';
+import { mapCorrectionReasonsAndValuesToFormattedValues } from './map-reasons-and-values-to-formatted-values';
+import { GetCompletedFeeRecordCorrectionsResponseBody } from '..';
+
+/**
+ * Maps an array of completed fee record corrections to a response format.
+ * @param completedCorrections - Array of fee record correction entities to be mapped
+ * @returns Array of mapped fee record corrections containing formatted values
+ * @throws Error if a completed correction has a null dateReceived value
+ */
+export const mapCompletedFeeRecordCorrectionsToResponse = (completedCorrections: FeeRecordCorrectionEntity[]): GetCompletedFeeRecordCorrectionsResponseBody =>
+  completedCorrections.map((feeRecordCorrection) => {
+    const { id, feeRecord, dateReceived: dateSent, reasons, previousValues, correctedValues, bankCommentary } = feeRecordCorrection;
+
+    if (dateSent === null) {
+      throw new Error('Invalid state: "dateReceived" is null but correction is marked as completed.');
+    }
```

Comment:
Minor comment but this looks pretty weird to be checking for a field and then naming a different thing in the error message. I think it might be best to rename at the bottom of the file when we return: `dateSent: dateReceived`